presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -1202,26 +1202,32 @@ public String toString()
             extends AbstractTupleDomainFilter
     {
         private final TupleDomainFilter[] filters;
+        private final boolean nanAllowed;
 
-        private MultiRange(List<TupleDomainFilter> filters, boolean nullAllowed)
+        private MultiRange(List<TupleDomainFilter> filters, boolean nullAllowed, boolean nanAllowed)
         {
             super(true, nullAllowed);
             requireNonNull(filters, "filters is null");
             checkArgument(filters.size() > 1, "filters must contain at least 2 entries");
 
             this.filters = filters.toArray(new TupleDomainFilter[0]);
+            this.nanAllowed = nanAllowed;
         }
 
-        public static MultiRange of(List<TupleDomainFilter> filters, boolean nullAllowed)
+        public static MultiRange of(List<TupleDomainFilter> filters, boolean nullAllowed, boolean nanAllowed)
         {
-            return new MultiRange(filters, nullAllowed);
+            return new MultiRange(filters, nullAllowed, nanAllowed);
         }
 
         @Override
         public boolean testDouble(double value)
         {
+            if (Double.isNaN(value)) {
+                return nanAllowed;
+            }
+
             for (TupleDomainFilter filter : filters) {
-                if (filter.testDouble(value)) {
+                if ((nanAllowed && Double.isNaN(value)) || filter.testDouble(value)) {
```

Comment:
@bhhari 
- `nanAllowed && Double.isNaN(value)` filter is redundant; NaN has been checked already
- ditto testFloat